src/apps/openfluid/WareTasks.cpp

```diff
@@ -572,20 +572,83 @@ int WareTasks::processPurge() const
     return error("missing or empty ware sources path");
   }
 
-  std::string BuildType = openfluid::utils::CMakeProxy::DefaultBuildType;
-  if (m_Cmd.isOptionActive("build-type") && !m_Cmd.getOptionValue("build-type").empty())
+  if (!m_Cmd.isOptionActive("build-type") && !m_Cmd.isOptionActive("build-version"))
   {
-    BuildType = m_Cmd.getOptionValue("build-type");
+    return error("missing build type or build version to purge");
   }
 
-  std::string buildPath = openfluid::tools::Filesystem::joinPath({ WarePath, 
-    openfluid::utils::CMakeProxy::getBuildDir(BuildType) });
+  std::string ReleaseModeStr = "release";
+  std::string DebugModeStr = "debug";
+  std::string CurrentVersionStr = "current";
+  std::string OtherVersionsStr = "other";
+
+  bool ReleaseMode = true;
+  bool DebugMode = true;
+  bool CurrentVersion = true;
+  bool OtherVersions = true;
 
-  if(!openfluid::tools::Filesystem::emptyDirectory(buildPath)) 
+  if(m_Cmd.isOptionActive("build-type") && !m_Cmd.getOptionValue("build-type").empty())
   {
-    return error("Error while clearing directory : " + buildPath);
+    std::vector<std::string> BuildTypes = openfluid::tools::split(m_Cmd.getOptionValue("build-type"), "+");
+    for(auto& BuildType : BuildTypes)
+    {
+      BuildType = openfluid::tools::toLowerCase(BuildType);
+      if(BuildType != ReleaseModeStr && BuildType != DebugModeStr)
+      {
+        return error("wrong build type");
+      }
+    }
+    if(std::find(BuildTypes.begin(), BuildTypes.end(), ReleaseModeStr) == BuildTypes.end())
```

Comment:
Since we have to iterate for two checks, a for loop would be more efficient here